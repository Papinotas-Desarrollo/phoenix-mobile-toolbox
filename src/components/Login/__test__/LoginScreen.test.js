import * as React from 'react';
import { TouchableOpacity } from 'react-native';
import renderer from 'react-test-renderer';
import { LoginScreen } from '../';

describe('Snapshot test', () => {
  it('With props snapshot', () => {
    const props = {
      source: {},
      bodyTitle: 'bodyTitle',
      bodySubtitle: 'bodySubtitle',
      beforeInputText: 'beforeInputText',
      firstInputPlaceholder: 'firstInputPlaceholder',
      secondInputPlaceholder: 'secondInputPlaceholder',
      onChangeEmail: jest.fn(),
      onChangePassword: jest.fn(),
      buttons: [<TouchableOpacity key="1" />, <TouchableOpacity key="2" />],
    };
    const tree = renderer.create(<LoginScreen {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Proptype check', () => {
  it('Callback proptypes defined', () => {
    expect(LoginScreen.propTypes.onChangeEmail).toBeDefined();
    expect(LoginScreen.propTypes.onChangePassword).toBeDefined();
  });
});
