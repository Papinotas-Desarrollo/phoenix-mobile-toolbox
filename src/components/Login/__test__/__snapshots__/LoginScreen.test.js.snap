// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test With props snapshot 1`] = `
<View
  onLayout={[Function]}
>
  <View
    style={
      Object {
        "bottom": 0,
      }
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#4A90E2",
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "minHeight": 70,
          "paddingHorizontal": 10,
          "paddingVertical": 20,
        }
      }
    >
      <Image
        source={Object {}}
      />
    </RCTSafeAreaView>
    <View
      style={
        Object {
          "backgroundColor": "#4A90E2",
          "flexDirection": "row",
          "justifyContent": "flex-start",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 22,
            "fontWeight": "600",
            "marginBottom": 13,
            "marginLeft": 13,
          }
        }
      >
        bodyTitle
      </Text>
    </View>
    <View>
      <Text
        style={
          Object {
            "color": "#FF9500",
            "fontSize": 20,
            "marginBottom": 5,
            "marginLeft": 13,
            "marginTop": 9,
          }
        }
      >
        bodySubtitle
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "marginBottom": 13,
            "marginHorizontal": 13,
          }
        }
      >
        beforeInputText
      </Text>
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        keyboardType="email-address"
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        placeholder="firstInputPlaceholder"
        rejectResponderTermination={true}
        style={
          Object {
            "fontSize": 16,
            "marginBottom": 10,
            "marginLeft": 13,
            "marginRight": 16,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
      />
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#D1D1D6",
              "borderBottomWidth": 2,
              "marginBottom": 42,
              "marginHorizontal": 13,
            },
            null,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginHorizontal": 13,
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          placeholder="secondInputPlaceholder"
          rejectResponderTermination={true}
          secureTextEntry={true}
          style={
            Object {
              "flex": 0.8,
              "fontSize": 16,
              "marginBottom": 10,
            }
          }
          underlineColorAndroid="transparent"
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "flex": 0.2,
            }
          }
        >
          <Text
            allowFontScaling={false}
            style={
              Array [
                Object {
                  "color": "#4A90E2",
                  "fontSize": 30,
                },
                null,
                Object {
                  "fontFamily": "Material Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            Óêó
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "borderBottomColor": "#D1D1D6",
              "borderBottomWidth": 2,
              "marginBottom": 42,
              "marginHorizontal": 13,
            },
            null,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row-reverse",
          "flexWrap": "wrap",
          "justifyContent": "space-between",
          "marginHorizontal": 10,
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      />
    </View>
  </View>
</View>
`;
